apps/modernization-ui/src/apps/search/investigation/InvestigationSearch.tsx

```diff
@@ -20,29 +21,13 @@ const InvestigationSearch = () => {
 
     const { terms } = useSearchResultDisplay();
 
-    const handleRemoveTerm = (term: Term) => {
-        const formValues = form.getValues();
-        const fieldNames = Object.keys(formValues);
-
-        const matchingField = fieldNames.find((fieldName) => fieldName === term.source);
-        if (matchingField && terms.length > 1) {
-            if (
-                matchingField === 'programAreas' ||
-                matchingField === 'jurisdictions' ||
-                matchingField === 'conditions'
-            ) {
-                form.setValue(
-                    matchingField,
-                    form.getValues()?.[matchingField]?.filter((p) => p.value !== term.value) ?? []
-                );
-            } else {
-                form.resetField(matchingField as keyof InvestigationFilterEntry);
-            }
-            search();
-        } else {
+    const handleRemoveTerm = removeTerm(form, () => {
+        if (terms.length === 1) {
             clear();
+        } else {
+            search();
         }
-    };
+    });
```

Comment:
suggestion:  This logic is copied in all of the search components.  If added [here](https://github.com/CDCgov/NEDSS-Modernization/blob/8b132319a172e295d7acc499a1785420c8fc778c/apps/modernization-ui/src/apps/search/layout/result/header/terms/SearchTerms.tsx#L23) then `onRemoveTerm` can be called only when there is more than one present.